lib/ansible/module_utils/basic.py

```diff
@@ -1804,17 +1805,19 @@ def _log_invocation(self):
         # TODO: generalize a separate log function and make log_invocation use it
         # Sanitize possible password argument when logging.
         log_args = dict()
-        passwd_keys = ['password', 'login_password', 'url_password']
 
         for param in self.params:
             canon  = self.aliases.get(param, param)
             arg_opts = self.argument_spec.get(canon, {})
             no_log = arg_opts.get('no_log', False)
+            arg_type = arg_opts.get('type', 'str')
 
             if self.boolean(no_log):
                 log_args[param] = 'NOT_LOGGING_PARAMETER'
-            elif param in passwd_keys:
+            # try to capture all passwords/passphrase named fields
+            elif arg_type == 'str' and PASSWORD_MATCH.search(param):
```

Comment:
Don't check arg_type here.  A module could do something silly like:
```
encryptor:
  passphrases_for_each_user:
   - foo
   - bar
   - baz
```
The arg_type in this case will be a list but we'd still want to catch it here.